LochNessBuilder/BuilderRegistry.cs

```diff
@@ -95,7 +95,7 @@ public static Builder<TInstance> Resolve<TInstance>() where TInstance : class, n
             var availableFactories = BuilderFactoryMethods[typeof(TInstance)];
```

Comment:
If there are no available factories appears we throw some dictionary key not found exception. Might be nicer to throw an exception with a nice error message. :)

Reply:
ahh, no, you just can't see the previous 3 lines where we do...
```
            if (!BuilderFactoryMethods.ContainsKey(typeof(TInstance)))
            {
                //Return a blank builder, which will just create `new TInstance()`.
                return Builder<TInstance>.New;
            }
```

Reply:
Ohh sorry! Quite right #fail

Reply:
Although I feel we probably want to throw an Exception if a builder wasn't found? (As using a new TInstance probably isn't what they were expecting - as they may just have set up their auto-builder wrong?)

Otherwise someone could just use WithFactory? Or we could even create a WithNew which explicitly does this?